src/raised-button.jsx

```diff
@@ -364,11 +369,11 @@ const RaisedButton = React.createClass({
 
     const iconCloned = icon && React.cloneElement(icon, {
       color: styles.label.color,
-      style: {
-        verticalAlign: 'middle',
-        marginLeft: labelPosition === 'before' ? 0 : 12,
-        marginRight: labelPosition === 'before' ? 12 : 0,
-      },
+      style: Object.assign(styles.icon, {
+        marginLeft: label && labelPosition !== 'before' ? 12 : 0,
+        marginRight: label && labelPosition === 'before' ? 12 : 0,
+        display: label || !linkButton ? 'inline-block' : 'block',
+      }),
```

Comment:
Please move these inside the styles.icon object so you wouldn't have to Object.assign.